company_default_partner_pricelist/models/res_company.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 ForgeFlow, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    default_property_product_pricelist_id = fields.Many2one(
+        "product.pricelist",
+        string="Default Account Pricelist",
+        help="Default pricelist for this company for new partners.",
+    )
+
+    def _update_partner_pricelist_generic_property(self):
+        self.ensure_one()
+        default_property_pp = self.default_property_product_pricelist_id
+        ir_property_obj = self.env["ir.property"].sudo()
+        field = self.env["ir.model.fields"]._get(
+            "res.partner", "property_product_pricelist"
+        )
+        ppty = ir_property_obj.sudo().search(
```

Comment:
could you help checking as if you already specify ir_property_obj with sudo, all the sudos on line 22, 33, 35 and 39 are not needed